src/Storages/FileLog/FileLogDirectoryWatcher.cpp

```
@@ -0,0 +1,88 @@
+    dw->itemMovedTo += Poco::delegate(this, &FileLogDirectoryWatcher::onItemMovedTo);
+}
+
+FileLogDirectoryWatcher::Events FileLogDirectoryWatcher::getEvents()
+{
+    std::lock_guard<std::mutex> lock(mutex);
+    Events res;
+    res.swap(events);
+    return res;
+}
+
+bool FileLogDirectoryWatcher::hasError() const
+{
+    return error;
+}
+
+const std::string & FileLogDirectoryWatcher::getPath() const
+{
+    return path;
+}
+
+void FileLogDirectoryWatcher::onItemAdded(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+{
+    std::lock_guard<std::mutex> lock(mutex);
+    DirEvent de;
+    de.callback = "onItemAdded";
+    de.path = ev.item.path();
+    de.type = ev.event;
+    events.emplace_back(de);
+}
+
+
+void FileLogDirectoryWatcher::onItemRemoved(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+{
+    std::lock_guard<std::mutex> lock(mutex);
+    DirEvent de;
+    de.callback = "onItemRemoved";
+    de.path = ev.item.path();
+    de.type = ev.event;
+    events.emplace_back(de);
+}
+
+
+void FileLogDirectoryWatcher::onItemModified(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+{
+    std::lock_guard<std::mutex> lock(mutex);
+    DirEvent de;
+    de.callback = "onItemModified";
+    de.path = ev.item.path();
+    de.type = ev.event;
+    events.emplace_back(de);
+}
+
+void FileLogDirectoryWatcher::onItemMovedFrom(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+{
+    std::lock_guard<std::mutex> lock(mutex);
+    DirEvent de;
+    de.callback = "onItemMovedFrom";
+    de.path = ev.item.path();
+    de.type = ev.event;
+    events.emplace_back(de);
+}
+
+void FileLogDirectoryWatcher::onItemMovedTo(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+{
+    std::lock_guard<std::mutex> lock(mutex);
+    DirEvent de;
+    de.callback = "onItemMovedTo";
+    de.path = ev.item.path();
+    de.type = ev.event;
+    events.emplace_back(de);
+}
+
+void FileLogDirectoryWatcher::onError(const Poco::Exception &)
+{
```

We shouldn't ignore the exception. May be at least log it.